spec/ddtrace/logger_spec.rb

```diff
@@ -0,0 +1,263 @@
+require 'spec_helper'
+
+require 'logger'
+require 'stringio'
+require 'time'
+require 'ddtrace'
+
+# Define functions to log to ensure the caller stack is consistent between tests
+def log_warn
+  Datadog::Tracer.log.warn('warn message')
+end
+
+def log_error
+  Datadog::Tracer.log.error('error message')
+end
+
+def log_debug
+  Datadog::Tracer.log.debug('debug message')
+end
+
+RSpec.describe Datadog::Logger do
+  default_logger = Datadog::Tracer.log
```

Comment:
These local variables will evaluate at load time which is probably not great; it might preempt initialization or hang onto some stale references.

Could we do this with an `around` block instead? Something like this would store the original, run the example, then restore the original after the test.

```
around do |example|
  Datadog::Tracer.log = Datadog::Tracer.log.tap do
    Datadog.configuration.logging_rate = Datadog.configuration.logging_rate.tap do
      example.run
    end
  end
end
```